view.php

```diff
@@ -20,32 +20,46 @@
 require_once($CFG->libdir.'/tablelib.php');
 
 $id = required_param('id', PARAM_INT);
+$download = optional_param('download', false, PARAM_ALPHA);
 
 list ($course, $cm) = get_course_and_cm_from_cmid($id, 'cpdlogbook');
 
 require_course_login($course, false, $cm);
 
 $record = $DB->get_record('cpdlogbook', [ 'id' => $cm->instance ], '*', MUST_EXIST);
 
+// If the table is being downloaded, then the $actions parameter is set to false.
+$table = new entries_table($cm, $USER->id, $OUTPUT, !$download, 'cpdlogbook_id');
+
+$table->is_downloading($download, 'cpdlogbook'.time(), 'cpdlogbook');
+
 $PAGE->set_url(new moodle_url('/mod/cpdlogbook/view.php', [ 'id' => $id ]));
-$PAGE->set_title($record->name);
-$PAGE->set_heading($record->name);
 
-echo $OUTPUT->header();
+if (!$download) {
+    // If the table is not being downloaded, render the normal page.
+
+    $PAGE->set_title($record->name);
+    $PAGE->set_heading($record->name);
+
+    echo $OUTPUT->header();
 
-echo html_writer::alist([
-        'id' => $record->id,
-        'course' => $record->course,
-        'name' => $record->name,
-        'totalpoints' => $record->totalpoints,
-        'info' => $record->intro,
-        'introformat' => $record->introformat,
-]);
+    echo html_writer::alist([
+            'id' => $record->id,
+            'course' => $record->course,
+            'name' => $record->name,
+            'totalpoints' => $record->totalpoints,
+            'info' => $record->intro,
+            'introformat' => $record->introformat,
+    ]);
 
-echo html_writer::link(new moodle_url('/mod/cpdlogbook/edit.php', ['id' => $id, 'create' => true]), 'Insert a record');
+    echo html_writer::link(new moodle_url('/mod/cpdlogbook/edit.php', ['id' => $id, 'create' => true]), 'Insert a record');
 
-$table = new entries_table($cm, $USER->id, $OUTPUT, 'cpdlogbook_id');
-$table->define_baseurl($PAGE->url);
-$table->out(40, true);
+    $table->define_baseurl($PAGE->url);
+    $table->out(40, true);
 
-echo $OUTPUT->footer();
+    echo $OUTPUT->footer();
+} else {
+    // If the table is being downloaded, only display the table.
+    $table->define_baseurl($PAGE->url);
```

Comment:
This duplicated define_baseurl line could be refactored up to where $table is created